### PR TITLE
*: add skip store limit option for scatter region

### LINKED ISSUE #6592: Scatter region will be restricted by the remove peer limit if the region is not empty
## Enhancement Task
Previously, we use light peer to skip the store limit check for the scatter region operator. But after we support the remove peer limit, we didn't take the light peer into consideration. When we scatter the non-empty region, the operator could be limited and directly canceled.
```
[2023/06/09 12:20:06.536 +08:00] [INFO] [cluster_worker.go:145] ["alloc ids for region split"] [region-id=1293] [peer-ids="[1294]"]
[2023/06/09 12:20:06.548 +08:00] [INFO] [cluster_worker.go:237] ["region batch split, generate new regions"] [region-id=222] [origin="id:1287 start_key:\"7480000000000081FFD95F72FFFFFFFFFFFFFFFFFF0000000000FB\" end_key:\"7480000000000081FFF25F698000000000FF000001FB00000000FB\" region_epoch:<conf_ver:25 version:191 > peers:<id:1288 store_id:10 > id:1289 start_key:\"7480000000000081FFF25F698000000000FF000001FB00000000FB\" end_key:\"7480000000000081FFF25F698000000000FF000002FB00000000FB\" region_epoch:<conf_ver:25 version:191 > peers:<id:1290 store_id:10 > id:1291 start_key:\"7480000000000081FFF25F698000000000FF000002FB00000000FB\" end_key:\"7480000000000081FFF25F698000000000FF000003FB00000000FB\" region_epoch:<conf_ver:25 version:191 > peers:<id:1292 store_id:10 > id:1293 start_key:\"7480000000000081FFF25F698000000000FF000003FB00000000FB\" end_key:\"7480000000000081FFF25F698000000000FF000004FB00000000FB\" region_epoch:<conf_ver:25 version:191 > peers:<id:1294 store_id:10 > id:1295 start_key:\"7480000000000081FFF25F698000000000FF000004FB00000000FB\" end_key:\"7480000000000081FFF25F72FFFFFFFFFFFFFFFFFF0000000000FB\" region_epoch:<conf_ver:25 version:191 > peers:<id:1296 store_id:10 >"] [total=5]
[2023/06/09 12:20:06.610 +08:00] [INFO] [operator_controller.go:611] ["operator canceled"] [region-id=1293] [takes=0s] [operator="\"scatter-region {mv light peer: store [10] to [4]} (kind:admin,region,leader, region:1293(191, 25), createAt:2023-06-09 12:20:06.610477352 +0
800 CST m=+1961.716996864, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, size:22, steps:[add learner peer 1299 on store 4, use joint consensus, promote learner peer 1299 on store 4 to voter, demote voter peer 1294 on store 10 to learner, transfer leader from store
 10 to store 4, leave joint state, promote learner peer 1299 on store 4 to voter, demote voter peer 1294 on store 10 to learner, remove peer on store 10]，timeout:[18m0s])\""] [additional-info=]
```
Related to https://github.com/pingcap/tidb/issues/44550

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #6592

### What is changed and how does it work?

This PR supports light peer for removing peer step. If we are doing scatter operation, there is an option to skip the store limit check.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by bufferflies — APPROVED** [review]

**Review by nolouch — COMMENTED** [review]
I wonder if the lightning scatter non-empty region, it's may cause tikv be busy and even influence the cluster if during the online import.

**Comment by 3pointer** [comment]
> I wonder if the lightning scatter non-empty region, it's may cause tikv be busy and even influence the cluster if during the online import.

According to my knowledge online import should only scatter the empty region. like restore a new table without any traffic. @lance6716 can you confirm this?

**Comment by 3pointer** [comment]
> > > I wonder if the lightning scatter non-empty region, it's may cause tikv be busy and even influence the cluster if during the online import.
> > 
> > 
> > According to my knowledge online import should only scatter the empty region. like restore a new table without any traffic. @lance6716 can you confirm this?
> 
> @nolouch one case is for parallel import (multiple lightning instances are running), lightning's progress is not synchronized to it will scatter non-empty region because other lightning instance has started ingestion.

but in this case cluster cannot serve traffic right? @lance6716

**Comment by JmPotato** [comment]
@3pointer @lance6716 @nolouch Any updates on this PR?

**Comment by 3pointer** [comment]
I think it should be caller's responsibility to set the store limit for different situation. so at least give an option of scatter region to let caller decide the limitation.

**Comment by nolouch** [comment]
I think it's ok to no limit in multi-rocks because the cost of generating a snapshot is relatively small. so add a option or directly judge in pd both ok to me.

File: pkg/schedule/operator/builder.go [hunk 1]
```diff
@@ -792,7 +792,11 @@ func (b *Builder) execRemovePeer(peer *metapb.Peer) {
 	if store != nil {
 		isDownStore = store.DownTime() > b.GetOpts().GetMaxStoreDownTime()
 	}
-	b.steps = append(b.steps, RemovePeer{FromStore: removeStoreID, PeerID: peer.GetId(), IsDownStore: isDownStore})
+	if b.lightWeight {
+		b.steps = append(b.steps, RemovePeer{FromStore: removeStoreID, PeerID: peer.GetId(), IsDownStore: isDownStore, IsLightWeight: b.lightWeight})
+	} else {
+		b.steps = append(b.steps, RemovePeer{FromStore: removeStoreID, PeerID: peer.GetId(), IsDownStore: isDownStore})
+	}
```
Comment (JmPotato) [inline pkg/schedule/operator/builder.go:799]: Why not just always assign `b.lightWeight`?

```suggestion
	b.steps = append(b.steps, RemovePeer{FromStore: removeStoreID, PeerID: peer.GetId(), IsDownStore: isDownStore, IsLightWeight: b.lightWeight})
```

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Review by 3pointer — APPROVED** [review]

**Pull request merged**